src/operator/tensor/indexing_op.cc

```diff
@@ -264,5 +264,33 @@ Examples::
 .add_argument("indices", "NDArray-or-Symbol", "array of locations where to set on_value")
 .add_arguments(OneHotParam::__FIELDS__());
 
+NNVM_REGISTER_OP(sparse_retain)
+.describe(R"code(pick rows specified by user input index array from a row sparse matrix
```

Comment:
Add an example here? Similar to the one in https://www.tensorflow.org/api_docs/python/tf/sparse_retain